server/handler/upgrade_handler.go

```diff
@@ -114,3 +121,96 @@ func (h ClusterUpgradeHandler) FinishUpgrade() (hasDone bool, err error) {
 	err = session.SyncNormalRunning(se)
 	return false, err
 }
+
+// SimpleServerInfo is some simple information such as version and address.
+type SimpleServerInfo struct {
+	infosync.ServerVersionInfo
+	ID           string `json:"ddl_id"`
+	IP           string `json:"ip"`
+	Port         uint   `json:"listening_port"`
+	JSONServerID uint64 `json:"server_id"`
+}
+
+// ClusterUpgradeInfo is used to report cluster upgrade info when do http request.
+type ClusterUpgradeInfo struct {
+	ServersNum          int                `json:"servers_num,omitempty"`
+	OwnerID             string             `json:"owner_id"`
+	UpgradedPercent     int                `json:"upgraded_percent"`
+	IsAllUpgraded       bool               `json:"is_all_server_version_consistent,omitempty"`
+	AllServersDiffInfos []SimpleServerInfo `json:"all_servers_diff_info,omitempty"`
+}
+
+func (h ClusterUpgradeHandler) showUpgrade(w http.ResponseWriter) error {
+	se, err := session.CreateSession(h.store)
+	if err != nil {
+		return err
+	}
+	defer se.Close()
+
+	// Check if we are upgrading by pausing user DDL.
+	isUpgrading, err := session.IsUpgradingClusterState(se)
+	if err != nil {
+		return err
+	}
+	if !isUpgrading {
+		WriteData(w, "The cluster state is normal. Guess that the upgrade has been completed, or that the upgrade did not use the method of pausing the user DDL.")
```

Comment:
What is "the method of pausing the user DDL"? I think it is confusing to users.

Reply:
It is the method of upgrading by pausing user DDL. That is, the `upgrade/start` mode is used.
Any suggestions?